### PR TITLE
Added automation providesCommand

### PR DESCRIPTION
**Pull request by tundisto**
## Status
Ready

## Related Issues
N/A

## Description
One question that frequently comes up both internally and with customers is "what integrations actually provide the command that's not enabled and causing my playbook to fail?"  This automation helps to answer that question.

## Screenshots
image

## Related PRs
N/A

## Required version of Demisto
4.x

## Does it break backward compatibility?
   No

## Must have
- [ ] Tests
- [ ] Documentation (with link to it)
- [ ] Code Review

## Dependencies
  Requires Demisto REST API integration be enabled.

## Additional changes
N/A

### CODE REVIEW

File: Scripts/script-providesCommand.yml [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+
+
+
+  try:
+      integration_commands = integration_commands_res[0]['Contents']['response']
+  except KeyError:
+      demisto.error('Did not receive expected response from Demisto API')
+      sys.exit()
+
+  #demisto.log(pformat(integration_commands))
```
Comment (Itay4) [inline Scripts/script-providesCommand.yml]: remove comment

File: Scripts/script-providesCommand.yml [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+      demisto.results(None)
+  else:
+      demisto.results(','.join(integrations_that_implement))
+type: python
+tags:
+- general
+comment: |-
+  Finds which integrations implement a specific Demisto command.  The results will be returned as comma-separated values (CSV).  The "Demisto API" integration must first be enabled.
+
+  This automation is S...L...O...W....
```
Comment (Itay4) [inline Scripts/script-providesCommand.yml]: ?

**Comment by Itay4** [comment]
@tundisto whats the status here?

**Comment by tundisto** [comment]
Converted to package format and merged master.

**Comment by Itay4** [comment]
@kirbles19 please review the yml

**Comment by Itay4** [comment]
@tundisto please note the [the build](https://circleci.com/gh/demisto/content/31322) fails

**Comment by tundisto** [comment]
It says "There is no test-playbook that checks the changes you've done".  Does that mean an actual playbook or a unit test?

**Comment by Itay4** [comment]
> It says "There is no test-playbook that checks the changes you've done". Does that mean an actual playbook or a unit test?

that refers to a test playbook, but a unit test should be fine in this case
just need to mention it under `tests` in the yml with something like `- No test - unit test`

**Comment by kirbles19** [comment]
@tundisto @Itay4 Done.

**Comment by tundisto** [comment]
@Itay4 unit test has been added and CI build is successful.  It's now complaining about CLA not being signed from a bunch of other commiters.

File: Scripts/ProvidesCommand/ProvidesCommand.py [hunk 3]
```diff
@@ -0,0 +1,78 @@
+import demistomock as demisto
+import sys
+# from pprint import pformat
```
Comment (Itay4) [inline Scripts/ProvidesCommand/ProvidesCommand.py]: remove this

File: Scripts/ProvidesCommand/ProvidesCommand.yml [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+name: ProvidesCommand
+script: ''
+type: python
+subtype: python3
+tags:
+- general
+comment: |-
+  Finds which integrations implement a specific Demisto command.  The results will be returned as comma-separated values (CSV).  The "Demisto REST API" integration must first be enabled.
+
+  This automation is S...L...O...W....
```
Comment (Itay4) [inline Scripts/ProvidesCommand/ProvidesCommand.yml]: remove this please

File: Scripts/ProvidesCommand/ProvidesCommand.py [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+            "uri": "/settings/integration/search",
+            "body": {"size": 1000}
+        }
+        integration_search_res = demisto.executeCommand("demisto-api-post", integration_search_args)
+        try:
+            integration_search = integration_search_res[0]['Contents']['response']
+        except KeyError:
+            demisto.error('Did not receive expected response from Demisto API')
+            sys.exit()
+        # demisto.log(pformat(integration_search))
```
Comment (Itay4) [inline Scripts/ProvidesCommand/ProvidesCommand.py]: remove these please

File: Scripts/ProvidesCommand/ProvidesCommand.py [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+        # sys.exit()
+    if integration_search:
+        integration_instances = integration_search['instances']
+        integration_instances_enabled = {}
+        for integration in integration_instances:
+            name = integration['brand']
+            # demisto.log(pformat(integration))
```
Comment (Itay4) [inline Scripts/ProvidesCommand/ProvidesCommand.py]: remove please

File: Scripts/ProvidesCommand/ProvidesCommand.py [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+            if integration['enabled'] == 'true':
+                integration_instances_enabled[name] = True
+
+    try:
+        integration_commands = integration_commands_res[0]['Contents']['response']
+    except KeyError:
+        demisto.error('Did not receive expected response from Demisto API')
+        sys.exit()
```
Comment (Itay4) [inline Scripts/ProvidesCommand/ProvidesCommand.py]: use `return_error` instead

File: Scripts/ProvidesCommand/ProvidesCommand.py [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+
+    # demisto.log(pformat(integration_commands))
```
Comment (Itay4) [inline Scripts/ProvidesCommand/ProvidesCommand.py]: remove please

File: Scripts/ProvidesCommand/ProvidesCommand.py [hunk 9]
```diff
@@ -0,0 +1,78 @@
...
+                if command_name == COMMAND:
+                    if ENABLED is None:
+                        integrations_that_implement.append(integration_name)
+                    elif ENABLED is True and integration_name in integration_instances_enabled:
+                        integrations_that_implement.append(integration_name)
+                    elif ENABLED is False and integration_name not in integration_instances_enabled:
+                        integrations_that_implement.append(integration_name)
+
+    if len(integrations_that_implement) == 0:
+        demisto.results(None)
```
Comment (Itay4) [inline Scripts/ProvidesCommand/ProvidesCommand.py]: prefer to return indicative message

File: Scripts/ProvidesCommand/ProvidesCommand.py (hunk 3 above)
Comment (tundisto) [reply Scripts/ProvidesCommand/ProvidesCommand.py]: done

File: Scripts/ProvidesCommand/ProvidesCommand.yml (hunk 4 above)
Comment (tundisto) [reply Scripts/ProvidesCommand/ProvidesCommand.yml]: done

File: Scripts/ProvidesCommand/ProvidesCommand.py (hunk 5 above)
Comment (tundisto) [reply Scripts/ProvidesCommand/ProvidesCommand.py]: done

File: Scripts/ProvidesCommand/ProvidesCommand.py (hunk 6 above)
Comment (tundisto) [reply Scripts/ProvidesCommand/ProvidesCommand.py]: done

File: Scripts/ProvidesCommand/ProvidesCommand.py (hunk 7 above)
Comment (tundisto) [reply Scripts/ProvidesCommand/ProvidesCommand.py]: done

File: Scripts/ProvidesCommand/ProvidesCommand.py (hunk 8 above)
Comment (tundisto) [reply Scripts/ProvidesCommand/ProvidesCommand.py]: done

File: Scripts/ProvidesCommand/ProvidesCommand.py (hunk 9 above)
Comment (tundisto) [reply Scripts/ProvidesCommand/ProvidesCommand.py]: done

**Comment by tundisto** [comment]
back to you, @Itay4

File: Scripts/ProvidesCommand/ProvidesCommand.py [hunk 10]
```diff
@@ -0,0 +1,74 @@
...
+        integration_instances_enabled = {}
+        for integration in integration_instances:
+            name = integration['brand']
+            if integration['enabled'] == 'true':
+                integration_instances_enabled[name] = True
+
+    try:
+        integration_commands = integration_commands_res[0]['Contents']['response']
+    except KeyError:
+        demisto.error('Did not receive expected response from Demisto API')
```
Comment (Itay4) [inline Scripts/ProvidesCommand/ProvidesCommand.py]: use return_error instead

Comment (tundisto) [reply Scripts/ProvidesCommand/ProvidesCommand.py]: done

**Comment by tundisto** [comment]
back to you, @Itay4

**Comment by tundisto** [comment]
this is ready for you, @Itay4

**Pull request merged**